qt/aqt/browser/table/state.py

```diff
@@ -6,21 +6,25 @@
 from abc import ABC, abstractmethod, abstractproperty
 from typing import List, Sequence, Union, cast
 
+from anki.browser import BrowserConfig
 from anki.cards import Card, CardId
-from anki.collection import Collection, Config
+from anki.collection import Collection
 from anki.notes import Note, NoteId
 from anki.utils import ids2str
 from aqt.browser.table import Column, ItemId, ItemList
 
 
 class ItemState(ABC):
-    config_key_prefix: str
     _active_columns: List[str]
-    _sort_column: str
-    _sort_backwards: bool
 
     def __init__(self, col: Collection) -> None:
+        notes_mode = self.is_notes_mode()
+        self.GEOMETRY_KEY_PREFIX = BrowserConfig.geometry_key_prefix(notes_mode)
+        self.SORT_COLUMN_KEY = BrowserConfig.sort_column_key(notes_mode)
+        self.SORT_BACKWARDS_KEY = BrowserConfig.sort_backwards_key(notes_mode)
```

Comment:
It feels a bit strange to be setting constants in an init method. If you made is_notes_mode() a  classmethod,  and moved the staticmethods from BrowserConfig into ItemState, making them also classmethods, then you could avoid having to pass in the bool each time, and we wouldn't need extra work in init. eg:

```
class ItemState:
    @classmethod
    def is_notes_mode() -> bool:
        # to be defined by subclasses
        assert False

    @classmethod
    def sort_key(cls) -> str:
        if cls.is_notes_mode():
            return "one"
        else:
            return "two"
```

Then we could call sort_key() on an instance of CardState or NoteState to get the correct key, and if add-ons modified BrowserConfig, those changes would be reflected in the returned values.

This approach of handling both cases in ItemState does feel a little atypical of OOP however - I think a more standard solution would be to override sort_key() in CardState and NoteState, instead of pushing that logic into the parent class.

Reply:
My earlier example of assigning directly to the constants instead of using a classmethod would have been even more straightforward - was there a problem with doing it that way that I overlooked? You could declare the same constants in ItemState with a dummy value and then set them correctly in CardState/NoteState

Reply:
My reasoning was, the more we handle in ItemState, the less duplicated code and the easier to read it. But I'll use your suggestion with the dummy variables. That means declaring the constants thrice, but at least we can keep the properties in ItemState.

> If you made is_notes_mode() a classmethod, and moved the staticmethods from BrowserConfig into ItemState, making them also classmethods

That wouldn't work, though, because the helpers are also used by `Collection._build_sort_mode()`, which is independent of any state.
I've left the helpers for now, but I can remove them if you want.

Reply:
Just realised I don't even need dummies, if I'm not mistaken.

Reply:
Thanks, I hadn't noticed _build_sort_mode. And just declaring the variables without dummies is nicer. Thanks Rumo!